v1/topdown/strings.go

```diff
@@ -514,18 +515,12 @@ func builtinSplit(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) e
 		return err
 	}
 
-	if !strings.Contains(string(s), string(d)) {
+	text, delim := string(s), string(d)
+	if !strings.Contains(text, delim) {
 		return iter(ast.ArrayTerm(operands[0]))
 	}
 
-	elems := strings.Split(string(s), string(d))
-	arr := make([]*ast.Term, len(elems))
-
-	for i := range elems {
-		arr[i] = ast.InternedTerm(elems[i])
-	}
-
-	return iter(ast.ArrayTerm(arr...))
+	return iter(ast.ArrayTerm(util.SplitMap(text, delim, ast.InternedTerm)...))
```

Comment:
I was just double checking that we're not _adding interned terms_ here. We're not. This is just looking up if we might be able to do without a new term for one that's already intermed, eh, interned.

Reply:
Indeed 🤝